src/igvfd/types/file_set.py

```diff
@@ -22,3 +22,39 @@ class FileSet(Item):
     base_types = ['FileSet'] + Item.base_types
     name_key = 'accession'
     schema = load_schema('igvfd:schemas/file_set.json')
+
+
+@collection(
+    name='analysis-sets',
+    unique_key='accession',
+    properties={
+        'title': 'Analysis sets',
+        'description': 'Listing of analysis sets',
+    })
+class AnalysisSet(FileSet):
+    item_type = 'analysis_set'
+    schema = load_schema('igvfd:schemas/analysis_set.json')
+
+    @calculated_property(
+        schema={
+            'title': 'Assay Title',
+            'description': 'Title(s) of assays that produced data analyzed in the analysis set.',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'title': 'Assay Title',
+                'description': 'Title of assay that produced data analyzed in the analysis set.',
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def assay_title(self, request, input_file_set=None):
+        assay_title = []
+        if input_file_set is not None:
+            for fileset in input_file_set:
+                file_set_object = request.embed(fileset, '@@object')
+                if file_set_object.get('assay_title') and \
+                        'MeasurementSet' in file_set_object.get('@type'):
+                    assay_title.append(file_set_object.get('assay_title'))
+            return list(set(assay_title))
```

Comment:
I think you can define assay_title as "set()", instead of appending you can use the method "add()" and then in the end you don't need to convert list into set and back into list, instead just convert set into list